examples/http/esxi_general_information.js

```diff
@@ -0,0 +1,267 @@
+ * @returns {Promise} A promise that resolves with the host reference extracted from the SOAP response.
+ */
+function fetchContainer(containerId) {
+  const payload = createSoapPayload(
+      '<Fetch xmlns="urn:vim25">' +
+      '    <_this type="ContainerView">' + containerId + '</_this>' +
+      '    <prop>view</prop>' +
+      '</Fetch>'
+  )
+  // Send the SOAP request and extract the host reference from the response.
+  return sendSoapRequest(payload, getHostRefFromSoap)
+}
+
+/**
+ * Constructs and sends a SOAP request to retrieve properties for a specified host reference.
+ * @param {string} hostRef - The reference ID of the host whose properties are to be retrieved.
+ * @returns {Promise} A promise that resolves with the properties of the host as extracted from the SOAP response.
+ */
+function retrieveProprieties(hostRef) {
+  const payload = createSoapPayload(
+      '<vim25:RetrieveProperties>' +
+      '   <vim25:_this type="PropertyCollector">ha-property-collector</vim25:_this>' +
+      '   <vim25:specSet>' +
+      '      <vim25:propSet>' +
+      '         <vim25:type>HostSystem</vim25:type>' +
+      '         <vim25:pathSet>runtime.connectionState</vim25:pathSet>' +
+      '         <vim25:pathSet>hardware.systemInfo.model</vim25:pathSet>' +
+      '         <vim25:pathSet>hardware.systemInfo.vendor</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.hardware.numNics</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.hardware.numHBAs</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.config.name</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.config.product.version</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.config.product.build</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.config.product.apiVersion</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.config.product.osType</vim25:pathSet>' +
+      '         <vim25:pathSet>summary.config.product.productLineId</vim25:pathSet>' +
+      '      </vim25:propSet>' +
+      '      <vim25:objectSet>' +
+      '         <vim25:obj type="HostSystem">' + hostRef + '</vim25:obj>' +
+      '      </vim25:objectSet>' +
+      '   </vim25:specSet>' +
+      '</vim25:RetrieveProperties>'
+  )
+  // Send the SOAP request and handle the response to generate the desired variables.
+  return sendSoapRequest(payload, generateVariables)
+}
+
+/**
+ * Parses the SOAP response and generates variables from the retrieved properties.
+ * @param {string} soapResponse - The SOAP response as a string.
+ * @returns {Array} An array of variables created from the extracted properties.
+ */
+function generateVariables(soapResponse) {
+  const $ = D.htmlParse(soapResponse);
+
+  // Helper function to extract the value of a property from the parsed response based on the given path.
+  function getPropSetValue(path) {
+    return $("propSet").has("name:contains('" + path + "')").find("val").text() || "N/A";
+  }
+
+  const variables = [
```

Comment:
put this as global variable and use it to generate the retreive properties envelope